### DISCUSSION TITLE
if MOOSE can be used with CFD

### DISCUSSION THREAD

**Discussion by DYLDYLDYL** [idaholab/moose#19989] [Q&A Modules: Navier-Stokes] [answered]
hello
I wonder if MOOSE can be used with CFD like openform？Because I need consider complex flow problems, two-equation turbulence model is required.
Thanks！

**Comment by WilkAndy** [comment]
Does the documentation at https://mooseframework.inl.gov/modules/navier_stokes/index.html answer your question?

**Reply by DYLDYLDYL** [reply]
Thanks！ There are many models used in CFD codes to compute the eddy viscosity. Some of the most popular are the isotropic 2-equation models, k-epsilon and k-omega. However, the MOOSE Navier-Stokes module only supports simple algebraic mixing-length models at present.
I need 2-equation. So I wonder if moose can couple with other CFD.

**Comment by GiudGiud** [comment, marked as answer]
Hello

MOOSE can be used with NekRS, an open source high performance CFD code, which can do LES and k-tau RANS (and others, but I m not an expert). All the transfers between Nek and MOOSE have already been written in Cardinal.

Cardinal is hosted here: https://github.com/neams-th-coe/cardinal

We currently do not have native 2-equation turbulence models in MOOSE and we only have plans to improve our coarse mesh 0-equation models currently. 

Guillaume

**Reply by DYLDYLDYL** [reply]
Thanks so much！
